src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SFA.DAS.AANHub.Domain.Entities;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+
+namespace SFA.DAS.AANHub.Data.Repositories;
+internal class CalendarEventsReadRepository : ICalendarEventsReadRepository
+{
+    private readonly AanDataContext _aanDataContext;
+    public CalendarEventsReadRepository(AanDataContext aanDataContext) => _aanDataContext = aanDataContext;
+
+    public async Task<CalendarEvent?> GetCalendarEvent(Guid id) => 
+        await _aanDataContext
```

Comment:
Can you capture the query that is being generated here please, I want to check if this is going to be inefficient to run when there is lot more data.

Reply:
Nice, not efficient query TBH.

Reply:
@DevenKShah We could consider using split queries to reduce the joins: 
https://learn.microsoft.com/en-us/ef/core/querying/single-split-queries